packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -511,11 +511,13 @@ export { create, create as update, read, destroy }
 
 export const config = {}
 
-export const all = async ({ root }) => {
+export const all = async ({ root }, processPlugins = true) => {
   const configSrc = await readConfigFile(root)
   const plugins = getPluginsFromConfig(configSrc)
 
-  return Promise.all(plugins.map(({ name }) => read({ root }, name)))
+  return Promise.all(
```

Comment:
I don't think this works with local plugins. I think it's a result of reading the config files and get the plugins, but I keep getting big errors like this:

```
Error: Cannot find module 'gatsby-plugin-test/package.json'
Require stack:
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/entry-74a70aa0.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/index.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/recipes.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.resolve (internal/modules/cjs/helpers.js:78:19)
    at listShadowableFilesForTheme (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/src/providers/gatsby/plugin.js:28:13)
    at read$4 (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/src/providers/gatsby/plugin.js:275:50)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 0)
    at _default (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/handlers/plugin.js:39:25)
    at Object.handler (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js:358:7) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/entry-74a70aa0.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/index.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/recipes.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/index.js'
  ]
}

 ERROR 

gatsby plugin <cmd> [plugins...]

Useful commands relating to Gatsby plugins

Positionals:
  cmd      Valid commands include `docs`, `ls`.
                      [string] [required] [choices: "docs", "ls"]
  plugins  The plugin names                 [array] [default: []]

Options:
  --verbose                Turn on verbose output
                                       [boolean] [default: false]
  --no-color, --no-colors  Turn off the color in output
                                       [boolean] [default: false]
  --json                   Turn on the JSON logger
                                       [boolean] [default: false]
  -h, --help               Show help                    [boolean]
  -v, --version            Show the version of the Gatsby CLI and
                           the Gatsby package in the current
                           project                      [boolean]


 ERROR 

There was an error


 ERROR 

Error: Cannot find module 'gatsby-plugin-test/package.json'
Require stack:
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/entry-74a70aa0.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/index.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/recipes.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js
- /Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.resolve (internal/modules/cjs/helpers.js:78:19)
    at listShadowableFilesForTheme (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/src/providers/gatsby/plugin.js:28:13)
    at read$4 (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/src/providers/gatsby/plugin.js:275:50)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 0)
    at _default (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/handlers/plugin.js:39:25)
    at Object.handler (/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js:358:7) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/entry-74a70aa0.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/node_modules/gatsby-recipes/dist/index.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/recipes.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/create-cli.js',
    '/Users/kyle/Github/gatsby/starters/hello-world/node_modules/gatsby-cli/lib/index.js'
  ]
}
```

I tried adding a local plugin with an empty package.json, and a filled out package.json, both gave this error. I got it outputting info from gatsby-source-filesystem. It's possible I'm doing something wrong testing this 🤔

Reply:
(we got this working so I'ma resolve the conversation, it seemed to be a problem with my setup somewhere)